### PR TITLE
[WNMGDS-2157] Interaction state VRT tests

### PR DESCRIPTION
**Pull request by forrestbaer**
## Summary

**NOTE: This changes the structure of the output of the snapshots. I will follow up with another PR that updates all snapshot names. Another ticket will be needed to do the work of figuring out a way to rename/name these in a better way. The only configuration option for this format in the playwright config is a little limited and some work will need to be done to possibly do this work in a global setup/teardown cycle.**

[https://jira.cms.gov/browse/WNMGDS-2157](https://jira.cms.gov/browse/WNMGDS-2157)

- Added another playwright config for interaction tests `tests/browser/interaction.config.ts`
- Playwright interaction tests are saved as `.test.interaction.ts` in the components folder, set up example tests for `Button`'s and `TextFields`.
- Updated existing tests to utilize the `themes.json` file in our root
- Updated `playwright/test` to `1.3.0`
- Updated README doc for VRT.

## How to test

1. `yarn clean && yarn install && yarn build && yarn build:storybook`
2. `yarn test:browser:interaction`
3. `yarn test:browser`
4. `yarn playwright install && yarn playwright test --config tests/browser/interaction.config.ts`

## Checklist

- [x] Prefixed the PR title with the [Jira ticket number](https://jira.cms.gov/projects/WNMGDS/) as `[WNMGDS-####] Title`

### If this is a change to code:

- [x] Created or updated unit tests to cover the logic added
- [x] If necessary, updated unit-test snapshots (`yarn test:unit:update`) and browser-test snapshots (`yarn test:browser:update`)
- [x] Checked for spelling and grammatical errors

### CODE REVIEW

**Comment by forrestbaer** [comment]
Spending some time fixing some things so CI will work..

File: package.json [hunk 1]
```diff
@@ -39,8 +39,10 @@
     "test:a11y": "yarn playwright test --config tests/a11y/playwright.config.ts",
     "test:unit": "yarn jest --config=tests/unit/jest.config.js",
     "test:unit:update": "yarn test:unit --updateSnapshot",
-    "test:browser": "docker run --rm --network host -v $(pwd):/work/ -w /work/ mcr.microsoft.com/playwright:v1.25.0-focal yarn playwright test --config tests/browser/playwright.config.ts",
-    "test:browser:update": "yarn build:storybook && docker run --rm --network host -v $(pwd):/work/ -w /work/ mcr.microsoft.com/playwright:v1.25.0-focal yarn playwright test --config tests/browser/playwright.config.ts -u",
+    "test:browser": "yarn test:browser:interaction && yarn test:browser:all",
+    "test:browser:all": "docker run --rm --network host -v $(pwd):/work/ -w /work/ mcr.microsoft.com/playwright:v1.30.0-focal yarn playwright test --config tests/browser/playwright.config.ts",
```
Comment (pwolfert) [inline package.json:43]: Are these command names flipped? Should `:all` be the one to run both the normal browser tests and the interaction tests?

File: tests/a11y/storybook.a11y.test.ts [hunk 2]
```diff
@@ -1,8 +1,7 @@
 import expectNoAxeViolations from './expectNoAxeViolations';
 import { test } from '@playwright/test';
 import { stories } from '../../storybook-static/stories.json';
-
-const themes = ['core', 'healthcare' /* 'medicare' */];
+import * as Themes from '../../themes.json';
```
Comment (pwolfert) [inline tests/a11y/storybook.a11y.test.ts:4]: Didn't we determine that `import *` inserts a `default` key? Why is it working?

File: tests/browser/interaction.config.ts [hunk 3]
```diff
@@ -0,0 +1,12 @@
+/**
+ * This config sets up interaction VRT tests in the codebase, we only need to
+ * change the testDir and testMap keys in the main playwright config.
+ */
+
+import { default as pwc } from './playwright.config';
```
Comment (pwolfert) [inline tests/browser/interaction.config.ts:6]: 👍

**Review by pwolfert — COMMENTED** [review]
This is really cool. Good work!

It's cool that we can see in the webkit screenshots that it doesn't apply the focus ring to clicked buttons. I found out from focus documentation that Sarah shared that Safari has different heuristics for [when to apply focus styles](https://developer.mozilla.org/en-US/docs/Web/CSS/:focus-visible). It's just neat to see that in our VRTs.

I'm currently getting failures when I run it due to tiny differences in text rendering. Did you capture these screenshots within docker or without?

File: package.json (hunk 1 above)
Comment (forrestbaer) [reply package.json:43]: That does make sense, I didn't want to mess with any CI changes.

File: tests/a11y/storybook.a11y.test.ts (hunk 2 above)
Comment (forrestbaer) [reply tests/a11y/storybook.a11y.test.ts]: Updated to use the format used everywhere else. Something to do with the way I was using it in the other place. I can't remember exactly now.

**Review by zarahzachz — COMMENTED** [review]
Code looks good, but when I run `yarn test:browser:interaction` I end up getting the following message in my terminal:

> Serving HTML report at http://localhost:59081. Press Ctrl+C to quit.
> Failed to open browser on http://localhost:59081

I've run it twice with the same sorta message (port numbers are different). Is this expected or am I not running something I should? I have Docker open/signed in.

**Review by pwolfert — APPROVED** [review]
LGTM

**Pull request merged**